### PR TITLE
refactor: update deno_fetch interface

### PR DESCRIPTION
**Pull request by bartlomieju**
Change deno_fetch::init to take parameter indicating
buildtime.

Add default implementation for FetchPermissions

### CODE REVIEW

File: op_crates/fetch/lib.rs [hunk 1]
```diff
@@ -58,9 +61,22 @@ pub fn init(isolate: &mut JsRuntime) {
   }
 }
 
+pub fn init_buildtime(isolate: &mut JsRuntime) {
+  init(isolate, true)
+}
+
+pub fn init_runtime(isolate: &mut JsRuntime) {
+  init(isolate, false)
+}
```
Comment (ry) [inline op_crates/fetch/lib.rs:70]: I think these methods are superfluous. Maybe just add a docstring comment to init explaining what the boolean flag does.

**Review by ry — APPROVED** [review]
Thanks!

**Review by ry — APPROVED** [review]
LGTM

File: op_crates/fetch/lib.rs [hunk 2]
```diff
@@ -46,7 +51,10 @@ pub fn init(isolate: &mut JsRuntime) {
   // workspace root.
   let display_root = manifest_dir.parent().unwrap().parent().unwrap();
   for file in files {
-    println!("cargo:rerun-if-changed={}", file.display());
+    if build_time {
+      println!("cargo:rerun-if-changed={}", file.display());
+    }
```
Comment (ry) [inline op_crates/fetch/lib.rs:56]: If we're initializing at runtime, ideally we'd use include_str for the source code...

File: op_crates/fetch/lib.rs [hunk 3]
```diff
@@ -33,34 +33,58 @@ use std::rc::Rc;
...
     let display_path = file.strip_prefix(display_root).unwrap();
     let display_path_str = display_path.display().to_string();
-    isolate
-      .execute(
-        &("deno:".to_string() + &display_path_str.replace('\\', "/")),
-        &std::fs::read_to_string(&file).unwrap(),
-      )
-      .unwrap();
+    let url = "deno:".to_string() + &display_path_str.replace('\\', "/");
+    isolate.execute(&url, source_code).unwrap();
```
Comment (douglascaetano) [inline op_crates/fetch/lib.rs:68]: Do you need to generate a string at runtime to get the full path of the file? Would not it be enough to have a static &str with crate name plus file name, like `deno_fetch/01_fetch_util.js`?

If so, file name at lines 40-59 could be generated at compile time.

Comment (ry) [reply op_crates/fetch/lib.rs]: Yes - it's a little annoying to have the paths hardcoded, but I agree that would be simpler. Updated.

**Review by bartlomieju — COMMENTED** [review]
LGTM

**Pull request merged**